xarray/backends/api.py

```diff
@@ -562,13 +566,32 @@ def open_mfdataset(paths, chunks=None, concat_dim=_CONCAT_DIM_DEFAULT,
 
     if lock is None:
         lock = _default_lock(paths[0], engine)
-    datasets = [open_dataset(p, engine=engine, chunks=chunks or {}, lock=lock,
-                             **kwargs) for p in paths]
-    file_objs = [ds._file_obj for ds in datasets]
 
+    open_kwargs = dict(engine=engine, chunks=chunks or {}, lock=lock,
+                       autoclose=autoclose, **kwargs)
+
+    if parallel or (parallel is None and get_scheduler() == 'distributed'):
+        # wrap the open_dataset, getattr, and preprocess with delayed
+        import dask
+        parallel = True
+        _open = dask.delayed(open_dataset)
```

Comment:
nit: per PEP8, we usually post-fix with an underscore for variables named to avoid conflicts with builtins, e.g., `open_` rather than `_open`.

Reply:
Thanks, wasn't aware of this detail.